032_tool.resolution_independent_qml.md

```diff
@@ -0,0 +1,47 @@
+# Resolution indepedent QML
+
+## Goals
+Creating UI code that can run on various display sizes with different resolutions
+
+## Description
+When creating a QtQuick application it is probable that its target device and its display will be completely different from the development machine.
+As a result, it is important to create the UI code in a way, that the UI is able to scale well across different display hardware.
+
+## Environment
+QML
+
+## Platform
+All
+
+## Implementation effort
+Medium
+
+## Applicability
+Always
+
+## Caveats
+- Resolution independence should be a goal from the start of the implementation phase
+- Converting a fixed resolution application to allow scaling can be costly
+
+## See also
+
+- [QML best practices](https://toolbox.basyskom.com/8)
+
+## Implementation hints
+Implementation is dependent on what kind of devices the application needs to run on. Some of the following questions can be used as guidance:
+
+- which types of devices need to be supported (example: smartwatch, mobile device, embedded device, desktop hardware, tv)
+- which resolutions need to be supported
+- which aspect ratios need to be supported
+
```

Comment:
additional points could be
- Should the application run in full screen or windowed mode?
- Should the application run in portrait or landscape mode and can you switch between them during runtime?